homeassistant/components/voice_assistant/websocket_api.py

```diff
@@ -173,3 +179,118 @@ def handle_binary(_hass, _connection, data: bytes):
         if unregister_handler is not None:
             # Unregister binary handler
             unregister_handler()
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "voice_assistant/pipeline/add",
+        vol.Required("conversation_engine"): str,
+        vol.Required("language"): str,
+        vol.Required("name"): str,
+        vol.Required("stt_engine"): str,
+        vol.Required("tts_engine"): str,
+    }
+)
+@websocket_api.async_response
+async def pipeline_add(
```

Comment:
Why are you not using [`StorageCollectionWebsocket`](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/collection.py#L432)?

Reply:
I don't like the way `StorageCollectionWebsocket` + `StorageCollection` mixes up the serialized representation of data with live data. 
It causes things like this, where data is transformed back and forth multiple times by voluptuous schemas to work around the `StorageCollection` design: https://github.com/home-assistant/core/blob/18babc2a3c23d34635bad978b94d04d9a96488db/homeassistant/components/schedule/__init__.py

Reply:
`StorageCollection` is improved in https://github.com/home-assistant/core/pull/90932, and this PR has been updated to use `StorageCollectionWebsocket` + `StorageCollection`